Makefile

```diff
@@ -1,7 +1,16 @@
+.PHONY = all testenv install package test test_in_containers test_in_container_37 test_in_container_38 clean container version
+
+HERE := $(shell pwd)
+HERE_OWNERSHIP := $(shell stat -c '%u:%g' $(HERE))
+FIX_OWNERSHIP := chown -R $(HERE_OWNERSHIP)
```

Comment:
Hmm. Instead of this should we be invoking the docker container with a non root uid .

Reply:
pattern I picked up forever ago, don't know why, yeah using a non-root uid is probably obviously the way to go

Reply:
having a user which is not in /etc/passwd is causing git to fall down during testing. In order to use --user in docker (which I think is the right way long term), I think I'll have to build an image. Not that prohibitive, especially as that feels like an additional issue to, in the future, move that image somewhere that it can be pulled rather than built on demand.

Thoughts/agreements/disagreements?